src/transactions/TransactionFrame.cpp

```diff
@@ -818,6 +818,31 @@ TransactionFrame::processFeeSeqNum(AbstractLedgerTxn& ltx, int64_t baseFee)
     {
         throw std::runtime_error("Unexpected database state");
     }
+
+    if (protocolVersionStartsFrom(header.current().ledgerVersion,
```

Comment:
We should perf test this:
* while this is correct, it computes things that are almost never used (at a minimum, we should not do this if there are `MergeOp` in the transaction set
* even if there are `bumpSeqOp` present, they only need as many source account as there are such operations

This may point to performing this at a higher level, maybe from `LedgerManagerImpl::processFeesSeqNums`, which would also allow to write logic using simpler datastructures (and working off a much smaller set of accounts) and only use `InternalLedgerEntry` at the end.

Reply:
I did some minor perf testing on my laptop and didn't notice anything, but I updated this using your suggestions. We could make this even more efficient (ex. only store the max seq for the specific accounts that will merge), but I don't think that's a good idea.